Description: This patch provides the generic architecture fallback, which was
 not merged by upstream. We just keep the patch here.
Author: Matthias Klose <doko@debian.org>
Last-Update: 2023-06-17
Bug-Debian: https://bugs.debian.org/1006920
Forwarded: https://github.com/oneapi-src/oneTBB/issues/776

Index: onetbb/src/tbb/tools_api/ittnotify_config.h
===================================================================
--- onetbb.orig/src/tbb/tools_api/ittnotify_config.h
+++ onetbb/src/tbb/tools_api/ittnotify_config.h
@@ -196,6 +196,10 @@
 #  define ITT_ARCH_RISCV64  10
 #endif /* ITT_ARCH_RISCV64 */
 
+#ifndef ITT_ARCH_GENERIC
+#  define ITT_ARCH_GENERIC 99
+#endif /* ITT_ARCH_GENERIC */
+
 #ifndef ITT_ARCH
 #  if defined _M_IX86 || defined __i386__
 #    define ITT_ARCH ITT_ARCH_IA32
@@ -217,6 +221,8 @@
 #    define ITT_ARCH ITT_ARCH_HPPA
 #  elif defined __riscv && __riscv_xlen == 64
 #    define ITT_ARCH ITT_ARCH_RISCV64
+#  else
+#    define ITT_ARCH ITT_ARCH_GENERIC
 #  endif
 
 #endif
